### ISSUE TITLE
No error message when when Rabbit MQ authentication fails

### ISSUE THREAD

**Issue by dkamenov** [sensu/sensu#1058]
When Sensu 0.20.3 server fails to log on to Rabbit MQ it does not produce an adequate error message in the logs to debug the issue. It just says "unsubscribing", even with `LOG_LEVEL=debug`

from sensu-server.log:

```
{"timestamp":"2015-09-22T03:58:53.442768+0000","level":"debug","message":"subscribing to keepalives"}
{"timestamp":"2015-09-22T03:58:53.443006+0000","level":"debug","message":"subscribing to results"}
{"timestamp":"2015-09-22T03:58:53.445557+0000","level":"warn","message":"reconnecting to transport"}
{"timestamp":"2015-09-22T03:58:53.445668+0000","level":"warn","message":"unsubscribing from keepalive and result queues"}
{"timestamp":"2015-09-22T03:58:53.445743+0000","level":"debug","message":"not currently the leader"}
```

**Comment by portertech** [comment]
@dkamenov the AMQP protocol is not very friendly as far as error reporting and messages, we can only guess if a connection failure is due to authentication. The AMQP lib does a best effort to determine when the failure occurred, during the connection process.

**Comment by dkamenov** [comment]
Perhaps that is true (I am not familiar with the protocol), but it looks like sensu knows when it can't bind to the queues, even if it does not know exactly why. Maybe a more descriptive error message, e.g "Could not bind to queues, check Rabbit MQ connection parameters". "Unsubscribing from queues" is just plain misleading when it couldn't subscribe in the first place.

In my case, I had installed sensu and dependencies on Amazon EC2, captured an AMI from that and spun up a new VM. On the new VM, the existing Rabbit MQ config was invalid (depends on hostname or something like that) and it wouldn't subscribe. As a workaround, I am adding the vhost/queue creation/set_permissions to the startup script

**Comment by anlutro** [comment]
I spent way too long debugging an issue because of this. Had to use `strace` to see the "authentication failure" message being returned from RabbitMQ.

**Comment by cwjohnston** [comment]
Hi @anlutro, I'm sorry to hear that you've spent so much time diagnosing RabbitMQ connection issues. We've just recently released Sensu 0.24 which includes upstream changes which should provide more context around AMQP connection errors. See discussion in https://github.com/sensu/sensu/issues/842 for more detail.

I think there's still some work to be done here however, as those new errors aren't logged. I've opened sensu/sensu-transport#30 to see if we can improve this situation.

**Comment by moises-silva** [comment]
@anlutro @dkamenov I've created PR sensu/sensu-transport#34 to address this issue. If you can try it and report back feedback that'd be great.

**Comment by cwjohnston** [comment]
Additional logging implemented in sensu/sensu-transport#34 will ship in Sensu 0.27

**Comment by portertech** [comment]
0.27.0-1 is in the unstable repositories, it fixes this issue. Will soon be promoted to stable.
